public/app/api/clients/provisioning/index.ts

```diff
@@ -13,26 +17,159 @@ import { createOnCacheEntryAdded } from './utils/createOnCacheEntryAdded';
   endpoints: {
-    listJob(endpoint) {
+    listJob: {
       // Do not include 'watch' in the first query, so we can get the initial list of jobs
       // and then start watching for changes
-      endpoint.query = ({ watch, ...queryArg }) => ({
+      query: ({ watch, ...queryArg }) => ({
         url: `/jobs`,
         params: queryArg,
-      });
-      endpoint.onCacheEntryAdded = createOnCacheEntryAdded<JobSpec, JobStatus, Job, JobList>('jobs');
+      }),
+      onCacheEntryAdded: createOnCacheEntryAdded<JobSpec, JobStatus, Job, JobList>('jobs'),
     },
-    listRepository(endpoint) {
-      endpoint.query = ({ watch, ...queryArg }) => ({
+    listRepository: {
+      query: ({ watch, ...queryArg }) => ({
         url: `/repositories`,
         params: queryArg,
-      });
-      endpoint.onCacheEntryAdded = createOnCacheEntryAdded<
-        RepositorySpec,
-        RepositoryStatus,
-        Repository,
-        RepositoryList
-      >('repositories');
+      }),
+      onCacheEntryAdded: createOnCacheEntryAdded<RepositorySpec, RepositoryStatus, Repository, RepositoryList>(
+        'repositories'
+      ),
+    },
+    deleteRepository: {
+      onQueryStarted: async (_, { queryFulfilled, dispatch }) => {
+        try {
+          await queryFulfilled;
+          dispatch(
+            notifyApp(
+              createSuccessNotification(
+                t(
+                  'provisioning.delete-repository-button.success-repository-deleted',
+                  'Repository settings queued for deletion'
+                )
+              )
+            )
+          );
+        } catch (e) {
+          if (e instanceof Error) {
+            dispatch(
+              notifyApp(
+                createErrorNotification(
+                  t('provisioning.delete-repository-button.error-repository-delete', 'Failed to delete repository'),
+                  e
+                )
+              )
+            );
+          }
+        }
+      },
+    },
+    deletecollectionRepository: {
+      onQueryStarted: async (_, { queryFulfilled, dispatch }) => {
+        try {
+          await queryFulfilled;
+          dispatch(
+            notifyApp(
+              createSuccessNotification(
+                t('provisioning.home-page.success-all-repositories-deleted', 'All configured repositories deleted')
+              )
+            )
+          );
+        } catch (e) {
+          if (e instanceof Error) {
+            dispatch(
+              notifyApp(
+                createErrorNotification(
+                  t('provisioning.home-page.error-delete-all-repositories', 'Failed to delete all repositories'),
+                  e
+                )
+              )
+            );
+          }
+        }
+      },
+    },
+    createRepositoryTest: {
+      onQueryStarted: async (_, { queryFulfilled, dispatch }) => {
+        try {
+          await queryFulfilled;
+          dispatch(notifyApp(createSuccessNotification('Test started')));
```

Comment:
test is synchronous -- so when it is done, the test has run.

I don't think we need a success note on this